### PR TITLE
Change arrayExpression->arrayOperand and array->arrayExpression.

### LINKED ISSUE #951: CQL2-JSON: "arrayExpression" was confusing
I believe that `minItems: 2` and `maxItems: 2` here is a left-over bug:

https://github.com/opengeospatial/ogcapi-features/blob/master/cql2/standard/schema/cql2.json#L258

### PR DESCRIPTION
**Pull request by pvretano**
Closes #951

### CODE REVIEW

File: cql2/standard/schema/cql2.json [hunk 1]
```diff
@@ -250,22 +250,22 @@
           "type": "string",
           "enum": ["a_containedBy", "a_contains", "a_equals", "a_overlaps"]
         },
-        "args": {"$ref": "#/$defs/arrayExpression"}
+        "args": {"$ref": "#/$defs/arrayOperand"}
       }
     },
-    "arrayExpression": {
+    "arrayOperand": {
```
Comment (jerstlouis) [inline cql2/standard/schema/cql2.json:256]: @pvretano I think this should be `arrayOperands` plural to be consistent with the other rules and to highlight the fact that this represents two operands for the arra predicate functions.

File: cql2/standard/schema/cql2.bnf [hunk 2]
```diff
@@ -178,16 +178,16 @@ temporalFunction = "T_AFTER"
 # a Boolean value of true.  Otherwise the function returns false.
 #=============================================================================#
 arrayPredicate = arrayFunction
-                 "(" arrayExpression "," arrayExpression ")";
+                 "(" arrayOperand "," arrayOperand ")";
```
Comment (jerstlouis) [inline cql2/standard/schema/cql2.bnf:181]: @pvretano Now I'm confused... Maybe this is is what was contributing to my earlier confusion.

Why does this take 2 `arrayOperand`?

File: cql2/standard/schema/cql2.json (hunk 1 above)
Comment (cportele) [reply cql2/standard/schema/cql2.json:256]: I agree, this should be plural. This applies also to cql2.yaml.

**Review by cportele — CHANGES_REQUESTED** [review]
In addition to changing "arrayOperand" in JSON/YAML to "arrayOperands", we also need to update the version information:

- In cql2.bnf we should add an entry in the history list at the top (which might be a placeholder until we tag a version for a 1.0.1 release).
- In cql2.yaml version should be "1.0.1-SNAPSHOT" (until we publish a 1.0.1).

**Comment by cportele** [comment]
Meeting 2024-10-07: @cportele will make the changes and then merge the PR.

**Pull request merged**